### PR TITLE
databases-week1-Shahnawaz

### PR DESCRIPTION
**Pull request by shazkhan2**

### CODE REVIEW

File: 5_databases/week1/command.sql [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+-- 2. Find out how many tasks in the task table do not have a valid due date
+
+SELECT id, due_date
+FROM task
+WHERE due_date IS NULL 
+   OR STR_TO_DATE(due_date, '%Y-%m-%d %H:%i:%s') IS NULL;
+
+
+
+-- 3. Find all the tasks that are marked as done
```
Comment (adionditsak) [inline 5_databases/week1/command.sql:16]: Awesome interpretation. I would probably just think more simple and check the status table first and check status_id=3 directly, but this works as well

File: 5_databases/week1/command.sql [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+SELECT status.name, user.id, task.description
+FROM status 
+left JOIN task ON status.id = task.status_id
+LEFT JOIN user ON user.id = task.user_id
+WHERE status.name = 'done';
+
+-- 4. Find all the tasks that are not marked as done
+SELECT status.name, user.id, task.description
```
Comment (adionditsak) [inline 5_databases/week1/command.sql:24]: same as above just with != (NOT equal operator)

File: 5_databases/week1/command.sql [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+FROM status 
+LEFT JOIN task ON status.id = task.status_id
+left JOIN user ON user.id = task.user_id
+WHERE status.name <> 'done';
+
+-- 5. Get all the tasks, sorted with the most recently created first
+SELECT status.name, created
```
Comment (adionditsak) [inline 5_databases/week1/command.sql:31]: There is no reason to join here. We just want the task:
```

SELECT * 
FROM task 
ORDER BY created DESC;

```

File: 5_databases/week1/command.sql [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+FROM task
+JOIN status ON task.status_id = status.id
+LEFT JOIN user ON user.id = task.user_id
+ORDER BY created;
+
+-- 6. Get the single most recently created task
+SELECT status.name, created
```
Comment (adionditsak) [inline 5_databases/week1/command.sql:38]: Same as above, keep it simple

File: 5_databases/week1/command.sql [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+FROM task
+JOIN status ON task.status_id = status.id
+LEFT JOIN user ON user.id = task.user_id
+ORDER BY created DESC LIMIT 1;
+
+-- 7. Get the title and due date of all tasks where the title or description contains database
+SELECT task.title, task.description
```
Comment (adionditsak) [inline 5_databases/week1/command.sql:45]: The task asks us to retrieve title and due_date, not description. Well done tho!

File: 5_databases/week1/command.sql [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+FROM task 
+WHERE task.title LIKE '%database%' OR task.description LIKE '%database%';
+
+-- 8. Get the title and status (as text) of all tasks
+SELECT task.title, status.name AS status_text
```
Comment (adionditsak) [inline 5_databases/week1/command.sql:50]: Well done!

File: 5_databases/week1/command.sql [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+--9. Get the name of each status, along with a count of how many tasks have that status
+SELECT status.name, COUNT(*) AS status_count
+FROM status
+JOIN task ON task.status_id = status.id
+GROUP BY status.name;
+
+-- 10. Get the names of all statuses, sorted by the status with most tasks first
+
+SELECT status.name, COUNT(*) AS status_count
```
Comment (adionditsak) [inline 5_databases/week1/command.sql:62]: Well done!

**Review by adionditsak — CHANGES_REQUESTED** [review]

File: 5_databases/week1/command.sql (hunk 4 above)
Comment (shazkhan2) [reply 5_databases/week1/command.sql:38]: updated as recommended!

**Review by adionditsak — APPROVED** [review]

**Comment by adionditsak** [comment]
@shazkhan2 I have approved this

**Pull request closed without merging**